src/shared/components/oncoprint/DeltaUtils.ts

```diff
@@ -870,28 +912,62 @@ function transitionHeatmapTrack(
                 : undefined
             )
         };
-        const newTrackId = oncoprint.addTracks([heatmapTrackParams])[0];
+        // register new track in oncoprint
+        const newTrackId:number = oncoprint.addTracks([heatmapTrackParams])[0];
+        // store relation between React heatmap track specs and OncoprintJS trackIds
         trackSpecKeyToTrackId[nextSpec.key] = newTrackId;
 
-        let trackIdForRuleSetSharingKey:"heatmap"|"heatmap01" = "heatmap";
-        if (nextSpec.molecularAlterationType === "METHYLATION") {
-            trackIdForRuleSetSharingKey = "heatmap01";
+        let trackIdForRuleSetSharingKey:"heatmap"|"heatmap01"|undefined;
+
+        switch (nextSpec.molecularAlterationType) {
+            case AlterationTypeConstants.TREATMENT_RESPONSE:
+
+                trackIdForRuleSetSharingKey = undefined;
+        
+                // register the new track id so that it will determine formatting of the track group
+                const rulesetTrackId = trackIdForRuleSetSharing.treatment![nextSpec.molecularProfileId];
+                if (rulesetTrackId) {
+                    oncoprint.shareRuleSet(newTrackId, rulesetTrackId);
```

Comment:
reverse arguments - the source track id goes first
 `shareRuleSet:(source_track_id:TrackId, target_track_id:TrackId)=>void;`

Reply:
Ah yes, you are right. This reverse argument does not do what I thought it would do. I would like to point out that there is a reverse logic for treatment profile tracks compared to others. I have looked at the code again and refactored to make this logic more clear. In essence the difference is:
- molecular profiles: a new molecular profile track is linked with the rule set of an existing track via the trackid of the existing set. If its the first track it is added to `trackIdForRuleSetSharing` so that others can link to it. The ruleSet association of existing tracks is never updated (link to the first track that was added). 
- treatment profiles: a new treatment profile track is always added to `trackIdForRuleSetSharing` so that others can link to it. The ruleSet association of existing treatment tracks is always updated to the track that was added last.

The reason for this different logic is that legends/ruleSets for treatments are based on an aggregation of data in all tracks in the trackGroup. The last added track holds up-to-date information on all tracks (includes itself). When this track is added the styling of all existing tracks are updated to point to the ruleSet defined by the last added track.

I think that this reverse logic in not necessary. When all track types would link to the ruleSet of the last added track irrespective of the data type this logic can be removed and the functionality is identical.

What do you think?

Reply:
Thank you for the clear explanation. Yes, I agree with your conclusion.

Reply:
So I'm still confused why you're only aggregating max/min over each profile, is that how it's done? You have a separate ruleset per-profile?

Reply:
No, the max and min values are evaluated per all tracks in a molecular profile (in DeltaUtils.ts):
```
    const treatmentProfilesMap = _.chain(nextProps.heatmapTracks)
                                    .filter((s:IHeatmapTrackSpec) => s.molecularAlterationType === AlterationTypeConstants.TREATMENT_RESPONSE)
                                    .groupBy((track:IHeatmapTrackSpec) => track.molecularProfileId)
                                    .mapValues( (o:IHeatmapTrackSpec[]) => _(o).flatMap((d)=>d.data).filter((d:IBaseHeatmapTrackDatum) => ! d.category).map((d)=>d.profile_data).value() )
                                    .value();

    // find the max and min treatment profile value in the next heatmap track group
    // max and min value is used to create a custom legend for the track group
    const treatmentProfileMaxValues = _.mapValues(treatmentProfilesMap, (profile_data:number[]) => { return _.max(profile_data) });
    const treatmentProfileMinValues = _.mapValues(treatmentProfilesMap, (profile_data:number[]) => { return _.min(profile_data) });
```
These values are attached to each trackspec and passed on to `transitionHeatmapTrack`:
```
        // add treatment layout/formatting information to the track specs
        track.maxProfileValue = treatmentProfileMaxValues[track.molecularProfileId];
        track.minProfileValue = treatmentProfileMinValues[track.molecularProfileId];

        transitionHeatmapTrack(track, prevHeatmapTracks[track.key], getTrackSpecKeyToTrackId,
            () => undefined, oncoprint, nextProps, {}, trackIdForRuleSetSharing, 
            undefined);
        delete prevHeatmapTracks[track.key];
```